jabgui/src/main/java/org/jabref/gui/newentry/NewEntryView.java

```diff
@@ -320,7 +320,16 @@ private void initializeLookupIdentifier() {
             Platform.runLater(() -> idLookupGuess.setSelected(true));
         }
 
-        idLookupGuess.selectedProperty().addListener((_, _, newValue) -> preferences.setIdLookupGuessing(newValue));
+        idLookupGuess.selectedProperty().addListener((_, _, newValue) -> {
+            preferences.setIdLookupGuessing(newValue);
+            // When switching to auto-detect mode, detect identifier type from current text
+            if (newValue && idText.getText() != null && !idText.getText().trim().isEmpty()) {
```

Comment:
null checks are from AI

did you check if this really happens????

You can use LOGGER for that